messaging/src/main/java/org/axonframework/messaging/responsetypes/ResponseType.java

```diff
@@ -39,20 +39,53 @@
  */
 public interface ResponseType<R> extends Serializable {
 
+    /**
+     * Indicates that the response type does not match with a given {@link java.lang.reflect.Type}.
+     */
+    int NO_MATCH = 0;
+
+    /**
+     * Indicates that the response matches with the {@link java.lang.reflect.Type}.while returning a single result.
+     */
+    int SINGLE_MATCH = 1;
+
+    /**
+     * Indicates that the response matches with the {@link java.lang.reflect.Type}.while returning an iterable result.
+     */
+    int ITERABLE_MATCH = 1024;
```

Comment:
I don't like the name here, it's revealing the intent - it is used only in `MultipleInstanceResponseType`. I'd rather have values like: `NO_MATCH`, `LOW_MATCH`, `HIGH_MATCH`, or smth within these lines.

Reply:
I can see what you mean. @smcvb Do you agree?

Reply:
Yep, I agree here. Let's rename the matching options.

I do think low/high might be a little too generic. I don't have something else to recommend at the moment, though.

Reply:
Alright, so we have too specific and too generic. How about a compromise as such?

On ResponseType:
- `MATCH`
- `NO_MATCH`

On MultipleInstanceResponseType:
- `ITERABLE_MATCH`

It's still a constant, still descriptive, but in the right place. In the end it translates to an `Integer`, so where it is defined does not matter for the `QueryBus`